src/components/product/product-name-creator.test.js

```diff
@@ -0,0 +1,59 @@
+import nameCreatorFactory, { stringifyStack } from './product-name-creator'
+
+describe('stringifyStack', () => {
+  it('verifies create string with product type', () => {
+    const string = stringifyStack([{ name: 'type', value: 'Pillow' }])
+    expect(string).toBe('Pillow')
+  })
+
+  it('verifies create string with product type and one option', () => {
+    const string = stringifyStack([
+      { name: 'type', value: 'Pillow' },
+      { name: 'type', value: 'size' },
```

Comment:
в реальности у нас не может быть двух объектов с одинаковым `name` тут. 
Давай вот так сделаем:
```suggestion
      { name: 'size', value: '1.5' },
```